### PR TITLE
Picker UI improvements

### PR DESCRIPTION
**Pull request by jmpressman**
## Changes

- Add customisable time picker visibility
- Add extract Single and DurationPicker logic into SingleUtilities and DurationUtilities
- Add more extensive tests

@rametta this should address the changes you requested from March. Apologies it took so long to get around to it. It's been a crazy year!

### CODE REVIEW

**Review by mschindlerMM — CHANGES_REQUESTED** [review]
I don't know why we're introducing the SingleUtilities and DurationUtilities module. We could just keep them in the SingleDatePicker / DurationDatePicker as the functions are never used outside of this module.

Unfortunately, this PR also leads to some breaking changes that I'm actually fine with. However, we should at least create a CHANGELOG.md file and describe the changes theere. Maybe even a simple upgrade guide to make it easier to apply the changes to our projects.

Otherwise, this PR looks pretty good to me. But there are a lot of changes and I might have missed something. So this PR should get at least one more review before we merge this.

**Comment by jmpressman** [comment]
> I don't know why we're introducing the SingleUtilities and DurationUtilities module. We could just keep them in the SingleDatePicker / DurationDatePicker as the functions are never used outside of this module.

I wanted to separate the code that defines the picker API & lifecycle (i.e. SingleDatePicker) from the code that is responsible for the business/data logic of the picker (i.e. SingleDatePickerUtilities). In my opinion the separation has made the code a lot easier to reason about as well as to determine what needs to be tested, two factors which should improve future maintainability.

> Unfortunately, this PR also leads to some breaking changes that I'm actually fine with. However, we should at least create a CHANGELOG.md file and describe the changes theere. Maybe even a simple upgrade guide to make it easier to apply the changes to our projects.

Happy to get a Changelog/upgrade guide in place.

File: src/SingleDatePicker.elm
Comment (jmpressman) [reply src/SingleDatePicker.elm:233]: @Tranquility The `timePickerVisibility` set in the settings is technically separate from the actual state of the time picker visibility. This isn't apparent for `NeverVisible` and `AlwaysVisible` which always map to `False` and `True`, respectively. But the `Toggleable ...` setting alone is not a sufficient indicator of whether or not the picker is open, it merely indicates it can be toggled.

To pass in the visibility setting itself would require us to move the open/closed state onto the `Toggleable` setting. This is not ideal because it a) would no longer be a static setting, b) exposes the time picker state directly to the user to manipulate instead of keeping it internal to the module.

**Comment by Tranquility** [comment]
Yes, I think moving them into a separate module allows us test functions without actually exposing them to the user so this sounds like a good idea to me.

In general it is really hard for me to comment on the changes without actually using it. But I realized that the changelog is the only place we have documented some of the settings right now. When reading the documentation for the settings we could do a better job of explaining what the the fields are for (https://package.elm-lang.org/packages/mercurymedia/elm-datetime-picker/latest/SingleDatePicker).

**Review by mschindlerMM — APPROVED** [review]

**Pull request merged**
